specs/preinstalls.md

```diff
@@ -0,0 +1,206 @@
+# Preinstalls
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [create2Deployer](#create2deployer)
+- [Safe](#safe)
+- [SafeL2](#safel2)
+- [MultiSend](#multisend)
+- [MultiSendCallOnly](#multisendcallonly)
+- [SafeSingletonFactory](#safesingletonfactory)
+- [MultiCall3](#multicall3)
+- [Arachnid's Deterministic Deployment Proxy](#arachnids-deterministic-deployment-proxy)
+- [Permit2](#permit2)
+- [ERC-4337 EntryPoint](#erc-4337-entrypoint)
+- [ERC-4337 SenderCreator](#erc-4337-sendercreator)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+[Preinstalled smart contracts](./glossary.md#preinstalled-contract-preinstall) exist on Optimism
+at predetermined addresses in the genesis state. They are  similar to precompiles but instead run
+directly in the EVM instead of running  native code outside of the EVM.
```

Comment:
```suggestion
at predetermined addresses in the genesis state. They are similar to precompiles but instead run
directly in the EVM instead of running native code outside of the EVM.
```